cou/apps/auxiliary.py

```diff
@@ -201,7 +201,7 @@ def pre_upgrade_steps(
         # action to clear the events before performing upgrade. If there
         # are no pending events, this step should be a no-op, so it's safe
         # to run anyway
-        if not self.config["enable-auto-restarts"].get("value"):
+        if self.config.get("enable-auto-restarts", {}).get("value") is False:
```

Comment:
Good point! From the [commit](https://opendev.org/openstack/charm-rabbitmq-server/commit/81c33953f918abca8d63fa80cf492d37c9326ce6#diff-0e4bc7aa994d87c68fe5fafe95babf22126792eb), the restart logic wasn't present in the charm. I am not sure how COU can help with that since there's not actions to perform the restart. 

On the other hand, if the user run COU the second time (the first time upgrades the charm to the version with this config option), then the restart can be properly performed, though we will miss one restart for the first time.

Reply:
oh, maybe not, pre upgrade and post upgrade has the same step, so if in pre upgrade the charm don't have the config, in post upgrade it will have it

Reply:
Good point, this need to be discussed with ManSol or tested.
> Here the logic suggests we're treating enable-auto-restarts=True as equivalent to the charm before the option was added.

Not sure if this is true, but I guess it is.

Reply:
I would doubt that we can test it locally, 3 years-old rabbitmq-server will probably does not work on focal at all, this mean COU will not support that too. Using 3.6/edge channel, I still get that config option, but COU is not supporting that already.

So before thinking whether `enable-auto-restarts` exists or not, we should consider if we want COU support very old release (not that ManSol will not have such environment).

Reply:
ManSol does have rabbitmq-server without the `enable-auto-restarts` but those are rabbitmq <= 3.6 which COU does not support in the first place. So we don't need to consider that case altogether.

Reply:
Resolving, since we have an answer